c/rc_lock_supply.h

```diff
@@ -0,0 +1,217 @@
+#ifndef _RC_LOCK_SUPPLY_H_
+#define _RC_LOCK_SUPPLY_H_
+#include "ckb_syscalls.h"
+
+typedef unsigned __int128 uint128_t;
+
+enum SupplyErrorCode {
+  ERROR_EXCEED_SUPPLY = 90,
+  ERROR_SUPPLY_AMOUNT,
+};
+
+// <1 byte version> <16 bytes current supply> <16 bytes max supply> <32 bytes sUDT script hash>
+#define MIN_INFO_CELL_LEN 65
+#define INFO_CELL_CONST_OFFSET 17
+
+typedef struct SupplyContextType {
+  size_t input_info_cell_index;
+  size_t input_info_cell_count;
+  size_t output_info_cell_index;
+  size_t output_info_cell_count;
+
+  uint8_t version;
+  uint128_t input_current_supply;
+  uint128_t output_current_supply;
+  uint128_t max_supply;
+  uint8_t sudt_script_hash[32];
+
+  uint128_t input_amount;
+  uint128_t output_amount;
+} SupplyContextType;
+
+int add_assign_amount(uint128_t* sum, const uint128_t* delta) {
+  *sum += *delta;
+
+  if (*sum < *delta) {
+    return ERROR_SUPPLY_AMOUNT;
+  } else {
+    return 0;
+  }
+}
+
+int minus_assign_amount(uint128_t* sum, const uint128_t* delta) {
```

Comment:
We may ignore the burn action in supply check? Since it does not need to be an owner to burn.

Reply:
@xxuejie , what's your suggestion?

Reply:
Fixed. return error code whiling burning.